pkg/trace/stats/aggregation.go

```diff
@@ -21,107 +24,41 @@ type Aggregation struct {
 	Env        string
 	Resource   string
 	Service    string
+	Type       string
 	Hostname   string
-	StatusCode string
+	StatusCode uint32
 	Version    string
 	Synthetics bool
 }
 
-// NewAggregationFromSpan creates a new aggregation from the provided span and env
-func NewAggregationFromSpan(s *pb.Span, env string) Aggregation {
-	synthetics := strings.HasPrefix(s.Meta[tagOrigin], "synthetics")
-
-	return Aggregation{
-		Env:        env,
-		Resource:   s.Resource,
-		Service:    s.Service,
-		Hostname:   s.Meta[tagHostname],
-		StatusCode: s.Meta[tagStatusCode],
-		Version:    s.Meta[tagVersion],
-		Synthetics: synthetics,
+func getStatusCode(s *pb.Span) uint32 {
+	strC := traceutil.GetMetaDefault(s, tagStatusCode, "")
+	if strC == "" {
+		return 0
+	}
+	c, err := strconv.Atoi(strC)
+	if err != nil {
+		log.Errorf("Invalid status code %s. Using 0.", strC)
```

Comment:
Is it the right level for an agent log? If the instrumentation is misconfigured this will flood the logs. From what I can see in other place most logging for invalid data is done with `log.Debugf`. @gbbr will know better than me.

Reply:
This is actually a redundant check as it's done before by the normalizer: https://github.com/DataDog/datadog-agent/blob/master/pkg/trace/agent/normalizer.go#L117-L122